app/src/main/java/com/rkm/tasky/ui/component/ReminderDropDownMenu.kt

```diff
@@ -0,0 +1,163 @@
+package com.rkm.tasky.ui.component
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.DropdownMenuItem
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.ExposedDropdownMenuBox
+import androidx.compose.material3.ExposedDropdownMenuDefaults
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.MenuAnchorType
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextField
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.Stable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.rkm.tasky.R
+import com.rkm.tasky.ui.theme.ItemDetailContainerColor
+import com.rkm.tasky.ui.theme.ItemDetailReminderDividerColor
+import com.rkm.tasky.ui.theme.ItemIconNotificationBackgroundColor
+import com.rkm.tasky.ui.theme.ItemIconNotificationBellColor
+import com.rkm.tasky.util.date.ReminderBeforeDuration
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun ReminderDropDownMenu(
+    state: ReminderDropDownMenuUiState,
+    actions: ReminderDropDownMenuActions,
+    modifier: Modifier = Modifier
+) {
+
+
+    ExposedDropdownMenuBox(
+        modifier = modifier,
+        expanded = state.isExpanded,
+        onExpandedChange = { actions.onExpanded() }
+    ) {
+        TextField(
+            modifier = Modifier.menuAnchor(MenuAnchorType.PrimaryNotEditable)
+                .fillMaxWidth(),
+            value = stringResource(state.selectedDuration.toUiString()),
+            onValueChange = {},
+            readOnly = true,
+            colors = ExposedDropdownMenuDefaults.outlinedTextFieldColors(
+                focusedBorderColor = ItemDetailReminderDividerColor,
+                unfocusedBorderColor = ItemDetailReminderDividerColor,
+            ),
+            leadingIcon = {
+                IconWithBackground(
+                    icon = {
+                        Icon(
+                            painter = painterResource(R.drawable.notications_icon),
+                            contentDescription = null,
+                            tint = ItemIconNotificationBellColor
+                        )
+                    },
+                    size = 30
+                )
+            }
+        )
+
+        ExposedDropdownMenu(
+            expanded = state.isExpanded,
+            onDismissRequest = { actions.onExpanded() },
```

Comment:
Isn't `onExpanded` the wrong wording here when it is about dismissing the menu?

Reply:
it's basically to update isExpanded lmao. Whether you click on something or out of it